enterprise/server/testutil/testredis/testredis.go

```diff
@@ -64,27 +70,48 @@ func Start(t testing.TB) string {
 	// ... but do break things if we reach the limit.
 	args = append(args, "--maxmemory-policy", "noeviction")
 	cmd := exec.CommandContext(ctx, redisBinPath, args...)
-	log.Printf("Starting redis server: %s", cmd)
+	log.Infof("Starting redis server: %s", cmd)
 	cmd.Stdout = &logWriter{}
 	cmd.Stderr = &logWriter{}
 	err = cmd.Start()
 	if err != nil {
-		assert.FailNowf(t, "redis binary could not be started", err.Error())
+		assert.FailNowf(h.t, "redis binary could not be started", err.Error())
 	}
-	var killed atomic.Value
-	killed.Store(false)
 	go func() {
-		if err := cmd.Wait(); err != nil && killed.Load() != true {
-			log.Warningf("redis server did not exit cleanly: %v", err)
+		if err := cmd.Wait(); err != nil {
+			close(h.done)
+			select {
+			case <-ctx.Done():
+			default:
+				log.Warningf("redis server did not exit cleanly: %v", err)
+			}
 		}
 	}()
-	t.Cleanup(func() {
-		log.Info("Shutting down Redis server.")
-		killed.Store(true)
-		cancel()
-	})
+}
+
+// Restart restarts Redis.
+// Blocks until Redis is reachable.
+func (h *Handle) Restart() {
+	h.cancel()
+	<-h.done
+	h.start()
+	waitUntilHealthy(h.t, h.Target)
+}
+
+// Start spawns a Redis server for the given test and returns a handle to the running process.
+func Start(t testing.TB) *Handle {
+	// redis socket must be in /tmp, redis won't read socket files in arbitrary locations
+	socketPath := testfs.MakeSocket(t, "redis.sock")
+	target := fmt.Sprintf("unix://%s", socketPath)
+
+	handle := &Handle{
+		Target:     target,
+		socketPath: socketPath,
```

Comment:
Does this need to pass `t: t,` ?

Otherwise LGTM

Reply:
Woops, good catch!